### PR TITLE
Added Mac OS Text To Speech Engine, some documentation updates

### PR DESCRIPTION
**Pull request by rudelm**
I've added the Mac OS Text To Speech Engine and added a readme for the audio generator.

I've also updated the documentation to give instructions on how to provide the firmware files with the SimpleHTTPServer

### CODE REVIEW

**Comment by dgiese** [comment]
Can someone confirm, that it does not break things? ;)

File: xiaomi.vacuum.gen1/audio_generator/audio_de.csv [hunk 1]
```diff
@@ -29,32 +29,32 @@ error16.wav,"Fehler 16, Benutzen Sie den Roboter auf einer flachen horizontalen
...
-sysupd_failed.wav,Firmwareupdate fehlgeschlagen. Verwende alte Version.
+power_off.wav,"Schalte aus."
+power_resume_clean.wav,"Batterie geladen. Setze Reinigung fort."
+remote_complete.wav,"Verwende Fernsteuerung."
+remote.wav,"Verwendung der Fernsteuerung abgeschlossen."
+restart_clean.wav,"Position unbekannt. Starte neue Reinigung."
+resume_clean.wav,"Setze Reinigung fort."
+resume_home.wav,"Kehre zurück zur Docking Station."
+return_no.wav,"Konnte nicht zu dem Startpunkt zurückkehren."
+return_yes.wav,"Konnte Dockingstation nicht finden. Platzieren Sie den Roboter näher an der Dockingstation und versuchen Sie es erneut.
```
Comment (fvollmer) [inline xiaomi.vacuum.gen1/audio_generator/audio_de.csv]: I'm not sure why you added all the quotes, but the ending quote is missing here. :)

Comment (rudelm) [reply xiaomi.vacuum.gen1/audio_generator/audio_de.csv]: Oh yes, thank you! I've wanted to have all lines in the same format as some of them were already using the quotes and some don't. This lead to some strange behaviour while the loop was executed.

File: xiaomi.vacuum.gen1/audio_generator/Readme.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Audio Generator for Xiaomi Vacuum Generation 1
+It is possible to create new language files which can be integrated into the rooted firmware. To generate the language files you can use the `generate_audio.py` script in this folder.
+
+It will read the content auf audio_xx.csv (whereby xx is a country language code like de_de or en_US) and use one of the supported engines to generate mp3 files for inclusion in the firmware. The language codes are the [ISO-639-1](https://en.wikipedia.org/wiki/ISO_639-1) format, so that the engine can use a voice with suitable pronounciation.
+
+You'll notice that each engine has its problems with mixed langauges (e.g. german with english words in it). A little testing is necessary to find a suitable pronounciation.
+
+# Requirements
+1. python3
+1. ffmpeg (to convert generated files into mp3)
```
Comment (fvollmer) [inline xiaomi.vacuum.gen1/audio_generator/Readme.md]: actually the gtts files are converted from mp3 to wav

Comment (rudelm) [reply xiaomi.vacuum.gen1/audio_generator/Readme.md]: you're right. Found this out once I've tested the default way with gtts. I first thought I would need ffmpeg as well on mac os but fortunately I could write them as wav. I'll create another PR to fix this in the readme

**Pull request merged**